enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+   as per point 1.c.ii and 1.c.iii is followed by the ingress operator. 
+
+#### 3. Update the existing EIP for the IngressController:
+   
+   When the admin changes the eip-allocations field in the Ingress Controller CR, the watch on the
+   `eip-allocations` field by the ingress operator scales a new deployment for the Ingress Controller CR.
+   Then step 1.c.iii is followed by the ingress operator.
+
+   
+### Risks and Mitigations
+
+Providing a right range of EIP as per AWS guidelines is an admin responsibility.
+If they are not provided the router won't get assigned an IP which can cause communication outage.
+As of now we are not able to identify any more risks which can affect other components of the EP.
+
+### Drawbacks
+
+Currently, there are not any drawbacks as such for this EP.
+
+## Open Questions
+
+1. Shall we support updating EIP in the NLB Ingress Controller ?
+
+Yes, we will support updating the EIP in the custom ingress controller.
+
+## Test Plan
+
+- This EP will be covered by unit tests.
+
+- This  EP will also be covered by end to end tests.
+  
+  The end to end test will cover the scenario where user will set an `eip-allocations` field
+  in the Ingress Controller CR and then test if the Ingress Controller
+  with service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+  has value of `eip-allocations` field from the Ingress Controller CR was created.
+
+  The end to end test will also the cover the scenario where user updates the `eip-allocations` field
+  in the Ingress Controller CR and then test if the Ingress Controller
+  with service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+  has value of `eip-allocations` field from the Ingress Controller CR was updated.
+  
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- N/A
+
+### Tech Preview -> GA
+
+- N/A
+
+### Removing a deprecated feature
+
+- N/A
+
+## Upgrade / Downgrade Strategy
+
+During upgrade, user should be able to populate the eip_allocations_field via installer
+
+During downgrade, the EIPs won't be allocated to the ingress controller's load balancer service.
+
+## Version Skew Strategy
+
+N/A
+
+## Operational Aspects of API Extensions
+
+- If the Ingress Operator can't scale an Ingress Controller with `eip-allocations` then
+  there can be an outage as all the communications to the router will fail.
+
+- In case of event of an outage because of this API the network edge team needs to be involved
+  in.
+
+## Support Procedures
+
+- If the service is not getting updated with the EIP annotation check the `kube-controller-manager` logs.
+  ```sh
+  oc logs <kcm pod name> -n openshift-kube-controller-manager
+  ```
+- If router pods fail to run then check the following:
+   - `Ingress Controller` logs.
+  ```sh
+  oc logs <ingress pod name> -n openshift-ingress
```

Comment:
Do you mean this?
```suggestion
  oc logs <router pod name> -n openshift-ingress
```

Reply:
yes. I think ingress is also right ?

Reply:
The term "ingress" is overloaded.  I would have asked for clarification if Candace hadn't.  Better to use "router pod" as it is the term we usually use is more obviously a reference to the pods that have "router" in the names.

Reply:
OK